app/controllers/users_controller.rb

```diff
@@ -138,7 +138,7 @@ def find_resource(permission = :view_users)
 
   def login_user(user)
     session[:user]         = user.id
-    uri                    = session[:original_uri]
+    uri                    = session.to_hash[:original_uri] || session.to_hash['original_uri']
```

Comment:
https://garrickvanburen.com/passing-sessions-and-referers-in-rails-functional-tests/
Old, but still true: In test, session keys are strings